enhancements/repositories-auto-pruning.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Repository Auto-pruning Policies
+authors:
+  - "@harishsurf"
+reviewers:
+  - "@bcaton85"
+  - "@dmage"
+approvers:
+  - "@bcaton85"
+  - "@dmage"
+creation-date: 2024-15-01
+last-updated: 2024-15-01
+status: implementable
+---
+
+# Repositories Auto-pruning Policies
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+In order for a container registry to run cost effectively unused resources should be deleted to reclaim storage space. This task should be done automatically in Quay by deleting unused tags through a given policy. The policy used should be configured by the user and a variety policy options should be given. Examples of initial policies would be prune tags older than X number of days or prune tags when more than 100 are present.
+
+## Motivation
+
+Reclaiming space is essential in running a cost efficient registry. Having this done automatically within Quay removes the need for manual intervention or custom software.
+
+### Goals
+
+* User is able to provide a time based policy for auto-pruning tags within a repository.
+* User is able to provide a tag number based policy for auto-pruning tags within a repository.
+* Namespace and repository policy are evaluated and a wider policy is applied.
```

Comment:
Suppose there's auto-prune policy at Org level as "keeping 10 tags", and there's auto-prune policy at Repo level as "keeping 5 tags", which one will apply to the repo, should be repo level policy, correct?

Reply:
Yes, always the wider policy which selects more tags to delete is applied. You can refer to the table in the enhancement. I have also tagged you in the google doc which has a table matrix with example values